onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -515,6 +516,14 @@ std::unique_ptr<IExecutionProvider> CreateExecutionProviderInstance(
             } else {
               ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_force_sequential_engine_build' should be a boolean i.e. 'True' or 'False'. Default value is False.\n");
             }
+          } else if (option.first == "trt_context_memory_sharing_enable") {
+            if (option.second == "True" || option.second == "true") {
+              params.trt_context_memory_sharing_enable = true;
+            } else if (option.second == "False" || option.second == "false") {
+              params.trt_context_memory_sharing_enable = false;
+            } else {
+              ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_context_memory_sharing_enable' should be a boolean i.e. 'True' or 'False'. Default value is False.\n");
```

Comment:
just noticed. I think we need to fix all these error strings.  
remove "a boolean i.e."
let's just have should be "True" or "False"   provider options are passed as strings. 
I can create another PR to fix. or you can fix if it's convenient to do so now.